### PR TITLE
[CI] Optimize workflows - skip runs on unrelated changes

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR accomplish?
- Adds additional guards and patterns in order to ignore running the most resource-heavy workflows on `master` and the release branches

Although we do have enough minutes to run this workflows, it doesn't change the fact that it's wasteful.

### CODE REVIEW

File: .github/workflows/backend-skipped-checks.yml [hunk 1]
```diff
@@ -4,12 +4,19 @@
...
 
 on:
+  push:
+    branches:
+      - 'master'
+      - 'release-**'
+    paths:
+      - "docs/**"
+      - "**.md"
+      - "**/frontend/test/**"
```
Comment (camsaul) [inline .github/workflows/backend-skipped-checks.yml:14]: Maybe this is a dumb question but can't we skip backend tests for frontend source changes as well? Frontend tests obviously don't affect backend stuff but I don't see any reason why something like `NemanjasReactComponent.jsx` would affect them either.

**Review by camsaul — APPROVED** [review]
Question

What happens if I have a long-running branch where I'm collaborating with someone else and we push both backend and frontend tests? Or maybe I update docs in a PR I'm working on? Presumably the dummy version of `be-linter-clj-kondo` gets triggered because of the matching path in addition to the normal version... does the normal one failing overwrite the results of dummy version succeeding? Is it a race condition? e.g. if the normal one fails and then dummy one gets triggered after do I get a false positive in CI?

I'm assuming this is already something you have figured out so this is more for my information than trying to poke holes in your PR. Assuming that doesn't do anything wacky then this looks good to me

We should try to figure out if we can skip more stuff and speed up CI or skip irrelevant tests... I'm going to jump off of a cliff if unrelated things keep flaking

File: .github/workflows/backend-skipped-checks.yml (hunk 1 above)
Comment (nemanjaglumac) [reply .github/workflows/backend-skipped-checks.yml:14]: That is an excellent question! @camsaul 
Tbh, I wasn't sure how coupled some things are but yes - you are right. If all backend tests are pure unit tests, there is no reason why frontend changes should affect them.

We probably need another layer of contract testing, but until then... E2E tests will do.

**Comment by nemanjaglumac** [comment]
> I'm assuming this is already something you have figured out so this is more for my information than trying to poke holes in your PR. Assuming that doesn't do anything wacky then this looks good to me

Great minds think alike :)
https://github.com/metabase/metabase/pull/25769#issuecomment-1271669439

Answer:
https://github.com/metabase/metabase/pull/25769#issuecomment-1272914434

**Pull request merged**